lib/ffi-compiler/compile_task.rb

```diff
@@ -130,27 +133,45 @@ def define_task!
           obj_file = obj_files[index]
           if src =~ /\.c$/
             file obj_file => [ src, File.dirname(obj_file) ] do |t|
-              sh "#{cc} #{cflags} -o #{t.name} -c #{t.prerequisites[0]}"
+              sh [cc, cflags, "-o", t.name, "-c", t.prerequisites[0]].shelljoin
+            end
+
+          elsif src =~ /\.go$/
+            file obj_file => [ src, File.dirname(obj_file) ] do |t|
+              sh [go, "build", "-buildmode=c-shared", "-o", t.name, t.prerequisites[0]].shelljoin
```

Comment:
You can simply use
```ruby
sh go, "build", "-buildmode=c-shared", "-o", t.name, t.prerequisites[0]
```
Shellwords is [incompatible to Windows](https://github.com/oneclick/rubyinstaller2/wiki/For-gem-developers#user-content-shell-escaping).

Reply:
ehhh, I assumed that Shellwords escapes for `cmd` on Windows...